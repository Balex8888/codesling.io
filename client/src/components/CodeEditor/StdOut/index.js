import React from 'react';

const StdOut = ({ text }) => {
  return (
    <div>
      {text.split('\n').map((singleLine, idx) => (
        <div key={`stdout-singleline-idx-${idx}`}>{singleLine}</div>
      ))}
    </div>
  );
};

export default StdOut;
