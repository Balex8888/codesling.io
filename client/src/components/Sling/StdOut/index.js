import React from 'react';

import Loading from '../../globals/Loading';

const StdOut = ({ text, loading }) => {
  return (
    <div className="stdout">
      {loading ? (
        <Loading />
      ) : (
        text.split('\n').map((singleLine, idx) => (
          <div key={`stdout-singleline-idx-${idx}`}>{singleLine}</div>
        ))
      )}
    </div>
  );
};

export default StdOut;
