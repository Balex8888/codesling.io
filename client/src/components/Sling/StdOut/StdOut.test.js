import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import StdOut from './index';

test('displays text passed in', () => {
  const component = shallow(
    <StdOut
      text="Hello, world!"
    />
  );
  expect(toJSON(component)).toMatchSnapshot();
});
