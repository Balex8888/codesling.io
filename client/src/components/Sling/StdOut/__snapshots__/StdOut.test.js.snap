// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays text passed in 1`] = `
<div
  className="stdout"
>
  <div
    key="stdout-singleline-idx-0"
  >
    Hello, world!
  </div>
</div>
`;
